Source/DafnyDriver/DafnyDoc.cs

```diff
@@ -0,0 +1,1362 @@
+using System;
```

Comment:
This file is too large. Could you extract things that are HTML but not Dafny specific? Have you considered using `HtmlTextWriter` ?

Also, I wonder whether you should move the AST node specific doc code into the AST nodes, using an interface like `ICanPrintHtmlDocs`. Otherwise how is this class going to be maintained? How does someone know to update this when this change the AST nodes?

I think ideally this class would even share code with the printer/grammar, because there is a lot of duplication between them, but that's far out of scope for now.

Reply:
Split it up to some extent. 15% smaller.

Reply:
You could also make the nodes partial so that all your code stays here.